src/util.test.js

```diff
@@ -0,0 +1,874 @@
+const jQuery = require('jquery');
+global.jQuery = jQuery;
+
+describe('throttle function executes given function ', () => {
+
+    test('once immediatelly and once after given wait interval when {leading: false} is not provided', async () => {
+
+        expect.assertions(2);
+
+        const functionCallCount = 5;
+        const wait = 500;
+        const mockFunction = jest.fn();
+
+        const throttledFunction = OskariMock.util.throttle(mockFunction, wait);
+
+        for (let i = 0; i < functionCallCount; i++) {
+            throttledFunction();
+            if (i === 0) {
+                expect(mockFunction.mock.calls.length).toEqual(1);
+            }
+        };
+        await sleep(wait);
+        expect(mockFunction.mock.calls.length).toEqual(2);
+    });
+
+    test('zero times immediatelly and once after given wait interval when {leading: false} is provided', async () => {
+
+        expect.assertions(2);
+
+        jest.setTimeout(7000);
+
+        const functionCallCount = 5;
+        const wait = 500;
+        const mockFunction = jest.fn();
+
+        const throttledFunction = OskariMock.util.throttle(mockFunction, wait, { leading: false });
+
+        for (let i = 0; i < functionCallCount; i++) {
+            throttledFunction();
+            if (i === 0) {
+                expect(mockFunction.mock.calls.length).toEqual(0);
+            }
+        };
+        /* Sleep enough to verify that only one function call within given wait time is set to be executed in future
+        *  (multiple throttle function calls received as burst within given wait- parameter do not spread all given function calls to be executed in future with interval of wait- parameter)
+        */
+        await sleep(wait * functionCallCount);
+        expect(mockFunction.mock.calls.length).toEqual(1);
+    });
+
+    test('once immediatelly and zero time after given wait interval when {trailing: false} is provided', async () => {
+
+        expect.assertions(2);
+
+        const functionCallCount = 2;
+        const wait = 500;
+        const mockFunction = jest.fn();
+
+        const throttledFunction = OskariMock.util.throttle(mockFunction, wait, { trailing: false });
+
+        for (let i = 0; i < functionCallCount; i++) {
+            throttledFunction();
+            if (i === 0) {
+                expect(mockFunction.mock.calls.length).toEqual(1);
+            }
+        };
+        await sleep(wait);
+        expect(mockFunction.mock.calls.length).toEqual(1);
+    });
+
+    test('immediatelly without throttling when called with interval >= throttle wait parameter', async () => {
+
+        const functionCallCount = 5;
+        expect.assertions(functionCallCount);
+
+        const wait = 200;
+        const mockFunction = jest.fn();
+
+        const throttledFunction = OskariMock.util.throttle(mockFunction, wait);
+
+        for (let i = 0; i < functionCallCount; i++) {
+            throttledFunction();
+            await sleep(wait);
```

Comment:
Move sleep and expect to another loop so calls are made immediately but execution should trickle to completion?

Reply:
Mentioned test intentionally tests that function should be called immediately when calling interval is equal to wait time provided to throttle function.
In other words tests that throttling should not occur when calling interval is equal to wait time.

According to test starting from https://github.com/mikkomustikkamaa/oskari-frontend/blob/add-first-unit-tests/src/util.test.js#L39, 
throttle function sets at most one function call to be executed with delay in situation where multiple function calls are issued before timeout of first function has elapsed. 
(Setting function execution with timeout is done only if timeout variable evaluate to false at line https://github.com/mikkomustikkamaa/oskari-frontend/blob/add-first-unit-tests/src/util.js#L65).

Reply:
Oh ok so the test is that if the call is NOT spammed the function WILL be called multiple times in contrast to other tests that spam calls and expect to only have it called once. That's what it says on the test description now that I read it with some thought :)